modules/social_features/social_group/social_group.module

```diff
@@ -2353,7 +2363,11 @@ function social_group_can_view_groups_of_type(string $type, AccountInterface $ac
   // permission. This counts for the advanced group permissions.
   // Admin, SM, CM.
   foreach ($user_roles as $role) {
-    $group_role = $group_role_synchroniser->getGroupRoleId($type, $role);
+    // group_role.synchronizer service has been removed, lets generates role_id
+    // directly.
+    $machine_name_max_length = EntityTypeInterface::ID_MAX_LENGTH - GroupTypeInterface::ID_MAX_LENGTH - 1;
+    $machine_name = substr(md5('group_role_sync.' . $role), 0, $machine_name_max_length);
+    $group_role = "$type-$machine_name";
```

Comment:
This feels a bit hacky now Group has decided to remove it I would expect there to be a better way? 

For example a lot of the new permission logic is now scope based as well, see:

```
  public function hasPermissionInGroup($permission, AccountInterface $account, GroupInterface $group) {
    $calculated_permissions = $this->groupPermissionCalculator->calculateFullPermissions($account);

    // First check if anything gave the user individual access to the group.
    $item = $calculated_permissions->getItem(PermissionScopeInterface::INDIVIDUAL_ID, $group->id());
    if ($item && $item->hasPermission($permission)) {
      return TRUE;
    }

    // Then check their synchronized access depending on if they are a member.
    if ($this->groupMembershipLoader->load($group, $account)) {
      $item = $calculated_permissions->getItem(PermissionScopeInterface::INSIDER_ID, $group->bundle());
    }
    else {
      $item = $calculated_permissions->getItem(PermissionScopeInterface::OUTSIDER_ID, $group->bundle());
    }

    return $item && $item->hasPermission($permission);
  }
```

Reply:
for this case we just needed to get a group role id to be able to proceed with access check,
previously we had following construction:
```
   // Check each role this user has whether its group counterpart has the correct
  // permission. This counts for the advanced group permissions.
  // Admin, SM, CM.
  foreach ($user_roles as $role) {
    $group_role = $group_role_synchroniser->getGroupRoleId($type, $role);
    if ($group_roles[$group_role]->hasPermission('view group')) {
      return TRUE;
    }
  }
```
So, as we already know that is only admins role (outsider scope), we can replace that whole check with the following:
```
    /** @var \Drupal\group\Access\GroupPermissionCalculatorInterface $group_permission_calculator */
    $group_permission_calculator = \Drupal::service('group_permission.calculator');
    $calculated_permissions = $group_permission_calculator->calculateFullPermissions($account);
    $item = $calculated_permissions->getItem(PermissionScopeInterface::OUTSIDER_ID, $group_type->id());

    if ($item && $item->hasPermission('view group')) {
      return TRUE;
    }
```

however, I would suggest moving any code refactoring  or improvements to a separate follow-up to prevent any possible regression bugs due to huge amount of changed source code

Reply:
already updated it